src/main/java/software/amazon/msk/auth/iam/IAMLoginModule.java

```diff
@@ -35,6 +36,7 @@ public class IAMLoginModule implements LoginModule {
     private static final Logger log = LoggerFactory.getLogger(IAMLoginModule.class);
 
     static {
+        ClassLoaderAwareIAMSaslClientProvider.initialize();
```

Comment:
* I must be making a mistake but I cannot find the `ClassLoaderAwareIAMSaslClientProvider` class in this change. Was it added in another or got left out?

* Also in order to reduce the risk of impact to customers who do not have multiple such class loaders, what do you think about creating a separate `IAMClassAwareLoginModule`. It could be used only by customers who use about multiple class loaders. Is the added complexity in configuration worth the risk mitigation?

Reply:
> I must be making a mistake but I cannot find the ClassLoaderAwareIAMSaslClientProvider class in this change. Was it added in another or got left out?

Good spot! For some reason your `.gitignore` has `internals` on there, meaning it was ignoring any new files within the `software.amazon.msk.auth.iam.internals` package. Will fix.

> Also in order to reduce the risk of impact to customers who do not have multiple such class loaders, what do you think about creating a separate IAMClassAwareLoginModule. It could be used only by customers who use about multiple class loaders. Is the added complexity in configuration worth the risk mitigation?

I think making this feature opt-in is a great idea to mitigate the risk, however I have the following concerns:
- Customers will not read the manual, and will not use the feature when they need to
- Customers will waste time root causing this problem and get frustrated
- I anticipate that the "leak" concern does not impact single classloader environments

Reply:
I agree that the tradeoffs favor building this into the existing `IAMLoginModule.`